extension/encoding/awslogsencodingextension/internal/unmarshaler/waf/unmarshaler.go

```diff
@@ -0,0 +1,206 @@
+	return &wafLogUnmarshaler{
+		buildInfo: buildInfo,
+	}
+}
+
+// See log fields: https://docs.aws.amazon.com/waf/latest/developerguide/logging-fields.html.
+type wafLog struct {
+	Timestamp           int64  `json:"timestamp"`
+	WebACLID            string `json:"webaclId"`
+	TerminatingRuleID   string `json:"terminatingRuleId"`
+	TerminatingRuleType string `json:"terminatingRuleType"`
+	Action              string `json:"action"`
+	HTTPSourceName      string `json:"httpSourceName"`
+	HTTPSourceID        string `json:"httpSourceId"`
+	HTTPRequest         struct {
+		ClientIP string `json:"clientIp"`
+		Country  string `json:"country"`
+		Headers  []struct {
+			Name  string `json:"name"`
+			Value string `json:"value"`
+		} `json:"headers"`
+		URI         string `json:"uri"`
+		Args        string `json:"args"`
+		HTTPVersion string `json:"httpVersion"`
+		HTTPMethod  string `json:"httpMethod"`
+		RequestID   string `json:"requestID"`
+		Fragment    string `json:"fragment"`
+		Scheme      string `json:"scheme"`
+		Host        string `json:"host"`
+	} `json:"httpRequest"`
+}
+
+func (w *wafLogUnmarshaler) UnmarshalLogs(content []byte) (plog.Logs, error) {
+	var errGzipReader error
+	gzipReader, ok := w.gzipPool.Get().(*gzip.Reader)
+	if !ok {
+		gzipReader, errGzipReader = gzip.NewReader(bytes.NewReader(content))
+	} else {
+		errGzipReader = gzipReader.Reset(bytes.NewReader(content))
+	}
+	if errGzipReader != nil {
+		if gzipReader != nil {
+			w.gzipPool.Put(gzipReader)
+		}
+		return plog.Logs{}, fmt.Errorf("failed to decompress content: %w", errGzipReader)
+	}
+	defer func() {
+		_ = gzipReader.Close()
+		w.gzipPool.Put(gzipReader)
+	}()
+
+	return w.unmarshalWAFLogs(gzipReader)
+}
+
+func (w *wafLogUnmarshaler) unmarshalWAFLogs(reader io.Reader) (plog.Logs, error) {
+	logs := plog.NewLogs()
+
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	resourceLogs.Resource().Attributes().PutStr(
+		string(conventions.CloudProviderKey),
+		conventions.CloudProviderAWS.Value.AsString(),
+	)
+
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(metadata.ScopeName)
+	scopeLogs.Scope().SetVersion(w.buildInfo.Version)
+
+	scanner := bufio.NewScanner(reader)
+	webACLID := ""
+	for scanner.Scan() {
+		logLine := scanner.Bytes()
+
+		var log wafLog
+		if err := gojson.Unmarshal(logLine, &log); err != nil {
+			return plog.Logs{}, fmt.Errorf("failed to unmarshal WAF log: %w", err)
+		}
+		if webACLID != "" && log.WebACLID != webACLID {
+			return plog.Logs{}, fmt.Errorf(
+				"unexpected: new web acl id %q is different than previous one %q",
+				webACLID,
+				log.WebACLID,
+			)
+		}
+		webACLID = log.WebACLID
+
+		record := scopeLogs.LogRecords().AppendEmpty()
+		if err := w.addWAFLog(log, record); err != nil {
+			return plog.Logs{}, err
+		}
+	}
+
+	if err := setResourceAttributes(resourceLogs, webACLID); err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to get resource attributes: %w", err)
+	}
```

Comment:
Should we do this only if `webACLID !- ""`? i.e. in case we get an empty log file. Not sure if that's a thing

Reply:
If it is empty, it will fail in the current application. Is that ok?

Reply:
I have added the check and a new unit test case for it.